go.mod

```diff
@@ -17,6 +17,7 @@ require (
 	github.com/palantir/witchcraft-go-tracing v1.17.0
 	github.com/stretchr/testify v1.8.1
 	golang.org/x/net v0.2.0
+	google.golang.org/protobuf v1.28.2-0.20221116083127-f0e23c7a8f8d
```

Comment:
Requires unreleased changes in the upstream `google.golang.org/protobuf` pkg.

Reply:
Hm this is going to apply to every project that uses CGR, including in unrelated protobuf usages. I don't love using a snapshot here. Do you know if the code generator also has to be upgraded to support the protodelim handlers? Or is it only the marshal/unmarshal functions?

Reply:
So using `promdelim` here allows us to stream [length-delimited](https://developers.google.com/protocol-buffers/docs/encoding#length-types) protobuf messages. These types of messages allow for streaming multiple probuf messages into one encoded payload. Given the way this is structured in CGR, I don't expect us to ever be encoding multiple probuf messages into one stream, so actually using `protodelim` I think was a mistake. 

I refactored to remove `promdelim` and also move us to a released version of `google.golang.org/protobuf`